integration_tests/steps/myco.py

```diff
@@ -55,3 +70,8 @@ def step_impl(context):
 @then("the myco client does not report errors")
 def step_impl(context):
     assert context.matcher.errors == 0
+
+
+@then("the api client's bid's/offer's attributes/requirements weren't violated")
+def step_impl(context):
+    assert context.aggregator.trade_violation_count == 0
```

Comment:
As I wrote in another comment, `context.aggregator.trade_violation_count` will fail if `context.aggregator` is not `test_stocks_attribute_requirement.AutoAggregator`. This means that the object we put in the context can't be re-used by other steps, and this is not optimal.
Since this attribute is very specific to a single class, I would suggest to create a step just for it. For example, you could just add this line inside the other step ("d3a-api-client is connected to simulation") and change its name. It would make it less flexible, but at least a bit clearer and safer to use.